api/bus.go

```diff
@@ -50,6 +55,12 @@ var (
 	}
 )
 
+// ArchiveContractsRequest is the request type for the /contracts/archive endpoint.
+type ArchiveContractsRequest struct {
+	ContractIDs []types.FileContractID `json:"contractIDs"`
+	Reason      string                 `json:"reason"`
```

Comment:
I feel like it might be more future proof to be able to specify a reason per contract. e.g. "failedChecks" could later be split up into the actual check that failed.

Reply:
I went back and forth on this myself actually... I added `failedchecks` because I figured you could consult the logs to see why it was archived anyway and passing in a reason per contract felt a bit clunky. I figured you could always group the contracts to archive per reason and send multiple requests...

I refactored it a bit and made it a map from fcid -> reason. Let me know what you think.